### PR TITLE
reduce the number of parameters that Process must be aware of

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

This is a refactor that makes the code easier to read because Process no longer needs to be aware of all the metadata parameters.

## Was the documentation (README, wiki) updated?
n/a

### CODE REVIEW

File: lib/sdr_client/deposit.rb [hunk 1]
```diff
@@ -10,8 +10,14 @@ def self.run(label: nil,
                  catkey:,
                  source_id:,
                  url:, files: [])
-      Process.new(label: label, type: type, url: url, files: files,
-                  apo: apo, collection: collection, catkey: catkey, source_id: source_id).run
+
+      metadata = Request.new(label: label,
```
Comment (justinlittman) [inline lib/sdr_client/deposit.rb]: Is this an HTTP Request? If so, seems like an odd way to be encapsulating method arguments.

File: lib/sdr_client/deposit/request.rb [hunk 2]
```diff
@@ -24,28 +24,50 @@ def initialize(label: nil,
...
       attr_reader :label, :uploads, :source_id, :catkey, :apo, :collection, :type
 
+      def administrative
+        {
+          hasAdminPolicy: apo
+        }
+      end
+
+      def identification
+        json = { sourceId: source_id }
```
Comment (mjgiarlo) [inline lib/sdr_client/deposit/request.rb]: In `#as_json` above you changed the structure from being like the one used now in `#identification` to one that uses `#tap`. Should this method use the same pattern? I don't have a clear favorite; either is fine by me. Or is there a reason to have them use different patterns I may be spacing on?

File: lib/sdr_client/deposit.rb (hunk 1 above)
Comment (jcoyne) [reply lib/sdr_client/deposit.rb]: no, it's a data model that represents a request in the abstract sense

File: lib/sdr_client/deposit/request.rb (hunk 2 above)
Comment (jcoyne) [reply lib/sdr_client/deposit/request.rb]: I like to consider myself a champion of diversity

**Pull request merged**